repos/system_upgrade/el7toel8/actors/sctpconfigread/actor.py

```diff
@@ -0,0 +1,159 @@
+from leapp.actors import Actor
+from leapp.models import SCTPConfig
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+
+#
+# Generic functions
+#
+def check_loaded_module(module):
+    """Check if a given module is loaded."""
+    import os
+    try:
+        os.stat('/sys/module/%s' % module)
+    except OSError:
+        logging.debug('Module %s is not loaded.' % module)
+        return False
+    logging.debug('Module %s is loaded.' % module)
+    return True
+
+def check_any_file_present(files):
+    import os
+    for f in files:
+        try:
+            if os.stat(f):
+                logging.debug('File %s is present.' % module)
+                return True
+        except OSError:
+            continue
+    return False
+
+#
+# Functions for probing for SCTP usage
+#
+def check_dlm_cfgfile():
+    """Parse the config file"""
+    fname = "test.conf"
+    try:
+        fp = open(fname, 'r')
+    except:
+        return False
+
+    cfgs = '[dlm]\n' + fp.read()
+    del fp
+
+    try:
+        from configparser import SafeConfigParser
+        cfg = SafeConfigParser()
+        cfg.read_string(fp)
+    except:
+        # Python2 ConfigParser doesn't have cfg.read_string
+        from cStringIO import StringIO
+        from ConfigParser import SafeConfigParser
+        cfg = SafeConfigParser()
+        cfg.readfp(StringIO(cfgs))
+
+    proto = cfg.get('dlm', 'protocol').lower()
+    if proto in ['sctp', 'detect', '1', '2']:
+        return True
+
+    return False
+
+def check_dlm_sysconfig():
+    """Parse /etc/sysconfig/dlm"""
+    try:
+        #fp = open('/etc/sysconfig/dlm', 'r')
```

Comment:
Timothy noticed I left a test code here. Will need a respin for this one at least.